core/primitives/src/child_trie.rs

```diff
@@ -183,15 +183,27 @@ impl ChildTrie {
 		&p[CHILD_STORAGE_KEY_PREFIX.len()..]
 	}
 	/// Constructor to use for building a new child trie.
-	/// This does not allow setting an existing `KeySpace`.
-	pub fn new(keyspace_builder: &mut impl KeySpaceGenerator, parent: &[u8]) -> Self {
-		let parent = Self::prefix_parent_key(parent);
-		ChildTrie {
-			keyspace: keyspace_builder.generate_keyspace(),
-			root: Default::default(),
-			parent,
-			extension: Default::default(),
-		}
+	///
+	/// By using a `KeySpaceGenerator` it does not allow setting an existing `KeySpace`.
+	/// If a new trie is created (see `is_new` method), the trie is not commited and
+	/// another call to this method will create a new trie.
+	/// 
+	/// This can be quite unsafe for user, so use with care (write new trie information
+	/// as soon as possible).
```

Comment:
in which context ? because you can create multiple ones in case you give wrong entry ?

Reply:
context is mainly the window of time where you create the child trie at a parent key but it is not updated (calling this function again will create another child trie instance at the same parent key which is not suitable).

Reply:
Yes ok I feel it is better that old new function as old new function would have been worse:

like you could do a new without fetching the current trie and then write on it and you would have two child trie with the same parent.